bitbot/reminder.go

```diff
@@ -0,0 +1,259 @@
+func joineventType(message *hbot.Message) string {
+	id, err := getMessageIDFromString(message.Content)
+	if err != nil {
+		return "Wrong command. format is : !remind join [ID]"
+	}
+
+	var event eventType
+
+	b.DB.Where("ID = ?", id).Take(&event)
+
+	if strings.Contains(event.People, message.Name) {
+		b.DB.Save(&event)
+		return "You already subscribed to this event"
+	}
+	event.People = fmt.Sprintf("%s%s ", event.People, message.Name)
+	b.DB.Save(&event)
+
+	feedback := fmt.Sprintf("Added %s to \"%s\"",
+		message.Name,
+		event.Description)
+
+	return feedback
+}
+
+func parteventType(message *hbot.Message) string {
+	id, err := getMessageIDFromString(message.Content)
+	if err != nil {
+		return "Wrong command. format is : !remind part [ID]"
+	}
+
+	var event eventType
+
+	b.DB.Where("ID = ?", id).Take(&event)
+
+	event.People = strings.Replace(event.People, message.Name+" ", "", -1)
+	b.DB.Save(&event)
+
+	feedback := fmt.Sprintf("Removed %s from \"%s\"",
+		message.Name,
+		event.Description)
+
+	return feedback
+}
+
+// Remove an event given by his ID
+func removeeventType(message *hbot.Message) string {
+	id, err := getMessageIDFromString(message.Content)
+	if err != nil {
+		return "Wrong command. format is : !remind remove [ID]"
+	}
+
+	var event eventType
+
+	b.DB.Where("ID = ? AND Author = ?", id, message.Name).Take(&event)
+
+	// Feedback Message construction
+	var feedbackMessage string
+	if event.ID == id {
+		feedbackMessage = fmt.Sprintf("Deleted event %d : %s",
+			event.ID,
+			event.Description)
+
+		// Delete
+		b.DB.Delete(&event)
+	} else {
+		feedbackMessage = "No event you own with that ID"
+	}
+
+	return feedbackMessage
+}
+
+// Lists all the awaiting events in PM
+func listeventTypes(message *hbot.Message, bot *hbot.Bot) string {
+	// Get all the db rows, iterate through them, format them and send them to pm
+	rows, err := b.DB.Model(&eventType{}).Rows()
+	if err != nil {
+		log.Error("Reminder: Couldn't get DB rows", err)
+	}
+
+	var (
+		event                   eventType
+		eventDescriptionMessage string
+	)
+
+	for rows.Next() {
+		err := b.DB.ScanRows(rows, &event)
+		if err != nil {
+			log.Error("Reminder: Couldn't get a db row", err)
+		}
+
+		eventDescriptionMessage = fmt.Sprintf(
+			"%d : [ %s ] at %s. eventType author : %s, in channel %s, with %s",
+			event.ID,
+			event.Description,
+			event.Time.Format(timeFormat),
+			event.Author,
+			event.Channel,
+			event.People)
+		bot.Msg(message.Name, eventDescriptionMessage)
+	}
+
+	return "I've PM'd you the list of awaiting events"
+}
+
+// Parses an event adding message and adds the event
+func addeventType(message *hbot.Message, bot *hbot.Bot) string {
+	// Parsing the message
+	channel := message.To
+	author := message.From
+	msg := strings.Split(message.Content, " ")
+	// Everything in the message content that isn't a command or a time
+	// specification is considered description
+	description := strings.Join(msg[2:len(msg)-2], " ")
```

Comment:
You should use `strings.SplitAfterN` here
https://golang.org/pkg/strings/#SplitAfterN